### PR TITLE
feat(infra): automated deploys

### PR DESCRIPTION
**Pull request by mr-uniswap**
This sets up a push-button deploy that uses our custom runner and token to deploy directly to NPM. 

This does not set the github tag on this repo automatically, and should probably be added later. This also does not set a deploy schedule

### CODE REVIEW

**Comment by mikeki** [comment]
I think we might have to create a `prepublish` script in the package.json to build SOR, it's been a common problem that SOR has to be manually built before deployment.

**Comment by mr-uniswap** [comment]
> I think we might have to create a `prepublish` script in the package.json to build SOR, it's been a common problem that SOR has to be manually built before deployment.

@mikeki the one nice thing with using npm ci is you rebuild it cleanly every time, so that won't be necessary unlike when you publish locally

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: Release
+
+on: workflow_dispatch # TODO set this on a deploy schedule
```
Comment (mr-uniswap) [inline .github/workflows/release.yml:3]: Note: this is so you can "press a button and deploy" later we change this to your deploy schedule

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+          registry-url: 'https://registry.npmjs.org'
+          # Defaults to the user or organization that owns the workflow file
+          scope: '@uniswap'
+      
+      - name: Build
+        run: npm ci
```
Comment (mr-uniswap) [inline .github/workflows/release.yml:37]: Npm ci will give you a clean build similar to `rm -rf build node_modules && npm install && npm run build`

File: .github/workflows/release.yml (hunk 1 above)
Comment (mikeki) [reply .github/workflows/release.yml:3]: Only Uniswap team members can click this button, right? :p

**Review by mikeki — APPROVED** [review]
LGTM let's also get Siyu's and maybe Christos opinion on the matter? but this would be way better DEVX

**Pull request merged**